src/vs/workbench/parts/files/common/explorerModel.ts

```diff
@@ -285,43 +285,53 @@ export class FileStat implements IFileStat {
 	 * Returns a child stat from this stat that matches with the provided path.
 	 * Will return "null" in case the child does not exist.
 	 */
-	public find(resource: URI): FileStat {
-
+	public find(resource: URI): ExplorerItem {
 		// Return if path found
-		if (resources.isEqual(resource, this.resource, !isLinux /* ignorecase */)) {
-			return this;
+		if (resource && this.resource.scheme === resource.scheme && this.resource.authority === resource.authority &&
+			startsWith(resource.path, this.resource.path)
+		) {
+			return this.findByPath(resource.path, this.resource.path.length);
 		}
 
-		// Return if not having any children
-		if (!this.children) {
-			return null;
+		return null; //Unable to find
+	}
+
+	public findByPath(path: string, index: number): ExplorerItem {
+		if (paths.isEqual(this.resource.path, path, !isLinux)) {
+			return this;
 		}
 
-		for (let i = 0; i < this.children.length; i++) {
-			const child = this.children[i];
+		if (this.children) {
+			while (index < path.length && path[index] === paths.sep) {
```

Comment:
@isidorn I understand the code below after some time but this could benefit from some comments what you are trying to achieve here